packages/datadog-instrumentations/src/router.js

```diff
@@ -149,6 +163,47 @@ function createWrapRouterMethod (name) {
         routeAddedChannel.publish({ topOfStackFunc: methodWithTrace, layer: this.stack.at(-1) })
       }
 
+      // Publish only if this router was mounted by app.use() (prevents early '/sub/...')
+      if (expressRouteAddedChannel.hasSubscribers && isAppMounted(this) && this.stack && this.stack.length > offset) {
+        // Handle nested router mounting for 'use' method
+        if (original.name === 'use' && arguments.length >= 2) {
+          const { mountPaths, startIdx } = extractMountPaths(arguments)
+
+          if (mountPaths.length) {
+            const parentPaths = getRouterMountPaths(this)
+
+            for (let i = startIdx; i < arguments.length; i++) {
+              const nestedRouter = arguments[i]
+
+              if (!nestedRouter || typeof nestedRouter !== 'function') continue
+
+              parentPaths.forEach(parentPath => {
+                mountPaths.forEach(normalizedMountPath => {
+                  const fullMountPath = joinPath(parentPath, normalizedMountPath)
+                  setRouterMountPath(nestedRouter, fullMountPath)
+                })
+              })
```

Comment:
Nit: Use for...of when possible, there's a slight performance improvement, but obviously in this case it doesn't matter much:

```suggestion
              for (const parentPath of parentPaths) {
                for (const normalizedMountPath of mountPaths) {
                  const fullMountPath = joinPath(parentPath, normalizedMountPath)
                  setRouterMountPath(nestedRouter, fullMountPath)
                }
              }
```